src/madd.cpp

```diff
@@ -7,38 +7,39 @@
 #include "renderer.h"
 #include "camera.h"
 #include "eventhandler.h"
-Madd::Madd(int width, int height, const char *title):mainCamera(NULL),
-                                                    close(false),
-                                                     width(width),
-                                                     height(height),
-                                                     timeScale(1.0f),
-                                                     lastFrame(Clock::now()){
-    render = new Renderer(this, width, height, title);
-    event = new EventHandler(this);
+
+void Madd::Init(int width, int height, const char *title) {
+	this->width = width;
+	this->height = height;
+	this->height = height;
+    Renderer::getInstance().Init(width, height, title);
     std::vector<unsigned int> keys = {KEY_ESCAPE,KEY_SPACE};
-    event->RegisterMultipleKeyCB(BIND(Madd::ProcessInput),keys);
+    EventHandler::getInstance().RegisterMultipleKeyCB(BIND(Madd::ProcessInput), keys);
+}
+
+Madd& Madd::getInstance() {
```

Comment:
I would prefer `Madd::GetInstance()` to fit the project style. Unless there's a reasoning behind making the function lowercase.